lib/pliny/db_support.rb

```diff
@@ -25,16 +25,19 @@ def initialize(url)
       end
     end
 
-    def migrate
-      Sequel::Migrator.apply(db, "./db/migrate")
+    def migrate(target=nil)
+      Sequel::Migrator.apply(db, "./db/migrate", target)
     end
 
     def rollback
+      return unless db.tables.include?(:schema_migrations)
+      return unless current = db[:schema_migrations].order(Sequel.desc(:filename)).first
+
       migrations = Dir["./db/migrate/*.rb"].map { |f| File.basename(f).to_i }.sort
-      current = db[:schema_migrations].order(Sequel.desc(:filename)).first[:filename].to_i
-      target = 0 # by default, rollback everything
-      if i = migrations.index(current)
-        target = migrations[i - 1] || 0
+      target     = 0 # by default, rollback everything
```

Comment:
Oh it's exactly what it does! it's just that it starts assuming it has to rollback to 0, and adjusts as it finds more recent migrations (just below).